### PR TITLE
Documentation and tools for simplifying configuration.

### PR DESCRIPTION
**Pull request by prescod**
Small documentation reorg and a new tool for partially validating settings.

### CODE REVIEW

File: docs/configuring.rst [hunk 1]
```diff
@@ -0,0 +1,139 @@
+===================
+Configuriing MetaCI
+===================
```
Comment (davisagli) [inline docs/configuring.rst:3]:
```suggestion
==================
Configuring MetaCI
==================
```

File: docs/configuring.rst [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+Some rough tests of whether these variables are set can be done using this
+command:
+
+    $ python manage.py verify_environment
+
+There are limits to how much validation can be done, however.
+
+Other Variables 
+---------------
```
Comment (davisagli) [inline docs/configuring.rst:100]: The vars in this section aren't really intended to be modified; I'd be inclined to take this section out.

File: docs/configuring.rst [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+REDIS_URL: 
+    This represents the url to the location where the redis server, configured for Meta CI. Set in Dockerfile.
+
+DJANGO_HASHID_SALT: 
+    This represents the hashid salt for the django application, currently set to 
+    arbritary string due to non production defaults, can be overridden 
+    in docker-compose.yml. Currently set in Dockerfile.
+    
+Build Arguments
```
Comment (davisagli) [inline docs/configuring.rst:128]: Clarify that these are only used when building the Docker image

File: metaci/users/management/commands/verify_environment.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+    def handle(self, *args, **options):
+        successes = []
+        errors = []
+
+        tests = [
+            (
+                scratch_org_limits,
+                "Devhub seems to be configured",
+                "Devhub is not configured. Scratch org creation will fail",
```
Comment (davisagli) [inline metaci/users/management/commands/verify_environment.py:25]:
```suggestion
                "Dev Hub seems to be configured",
                "Dev Hub is not configured. Scratch org creation will fail",
```

File: metaci/users/management/commands/verify_environment.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            ),
+            (
+                test_github,
+                "Github API is configured",
+                "Github API is not configured. Repos cannot be retrieved for builds.",
```
Comment (davisagli) [inline metaci/users/management/commands/verify_environment.py:30]:
```suggestion
                "GitHub API is configured",
                "GitHub API is not configured. Repos cannot be retrieved for builds.",
```

File: metaci/users/management/commands/verify_environment.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            ),
+        ]
+
+        for test in tests:
+            func, good, bad = test
+            try:
+                func()
+                successes.append(good)
+            except BaseException as e:
+                errors.append(bad + f": {e}")
```
Comment (davisagli) [inline metaci/users/management/commands/verify_environment.py:50]:
```suggestion
                errors.append(f"{bad}: {e}")
```

File: metaci/users/management/commands/verify_environment.py [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+            for error in errors:
+                print("ERROR:", error)
+            print()
+            print("There were errors:")
+            print("Check your .env file or environment.")
+            print("Please read configuring.rst for more details.")
+
+
+def test_connected_app():
+    # is there a way to test this with no user interaction?
```
Comment (davisagli) [inline metaci/users/management/commands/verify_environment.py:69]: Not without the instance_url and username of a user that has already approved the connected app

**Review by davisagli — CHANGES_REQUESTED** [review]

File: docs/configuring.rst [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+MetaCI must authenticate with the GitHub API to fetch repositories and create releases. 
+This can be set up for a GitHub user by setting GITHUB_USERNAME and GITHUB_PASSWORD, 
+or for a GitHub App by setting GITHUB_APP_ID and GITHUB_APP_KEY.
+
+GITHUB_USERNAME:     
+    This represents the username of either the tester or service account configured for MetaCI
+
+GITHUB_PASSWORD:      
+    This represents the password or personal access token a user must have to access 
+    their account a `personal access token` will be used when Multi Factor Authentication is enabled.
```
Comment (Br4nd0R) [inline docs/configuring.rst:41]:
```suggestion
    their account. A `personal access token` will be used when Multi Factor Authentication is enabled.
```

**Review by Br4nd0R — APPROVED** [review]
lgtm!

**Comment by prescod** [comment]
Unfortunately I can't see what Snyk dislikes about my build. Is it an orthogonal dependency problem?

**Comment by Br4nd0R** [comment]
> Unfortunately I can't see what Snyk dislikes about my build. Is it an orthogonal dependency problem?

I'm having the same problem on a Dependency PR for MetaDeploy. It is not clear at all what Snyk is actually complaining about, it tells me a file that it doesn't like but I can't find a specific error message. And I don't have access on this repo to confirm if this is the same thing I'm seeing. I'll let you know if I find out anything.

**Comment by davisagli** [comment]
@Br4nd0R You should have access to see MetaCI in snyk, from what I can see. @prescod, your status shows as "pending"; I think may have just sent you an invite that you have to accept but the UI is confusing.

In this case it looks clear to me what it's complaining about -- there are 2 dependencies which need to be updated. But they aren't related to this PR and I think we can go ahead and merge this and then update them separately. @Br4nd0R the problem in MetaDeploy where it wasn't showing the actual problem turned out to be a problem parsing base.txt because it listed the same package twice -- I figured it out because the CircleCI build was not running due to needing to be reset after we enabled SSO, and then once it ran it showed the error reading base.txt

**Review by davisagli — APPROVED** [review]
Thanks!

**Pull request merged**